pkg/kubelet/container/cache.go

```diff
@@ -103,12 +101,11 @@ func (c *cache) Set(id types.UID, status *PodStatus, err error, timestamp time.T
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) {
-		// Set the value in the cache only if it's not present already
-		// or the timestamp in the cache is older than the current update timestamp
-		if cachedVal, ok := c.pods[id]; ok && cachedVal.modified.After(timestamp) {
-			return false
-		}
+	// Added after Evented PLEG was added
```

Comment:
@pacoxu should it be gated by pleg.IsEventedPLEGInUse() (since we are removing `utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG)` and it was added after Evented PLEG?

Reply:
It will also maintain the existing behavior intact for Generic PLEG.

Reply:
there will be an import cycle 😓

Reply:
In that case, IMO it would make sense to keep `utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG)`  because otherwise it alters the behavior in existing Generic PLEG and we don't want to risk that.